src/modelterms.jl

```diff
@@ -198,9 +198,9 @@ end
 function Base.sparse(A::VectorFactorReTerm{T,V,R,S}) where {T,V,R,S}
     n = size(A, 1)
     colind = Matrix{Int32}(S, n)
-    rr = A.f.refs
+    lvls = levels(A.f)
     @inbounds for j in 1:n
-        offset = (rr[j] - 1) * S
+        offset = (findfirst(lvls,A.f[j]) - 1) * S
```

Comment:
I don't understand that code, but if `n` is large it would be more efficient to do something like `ord = CategoricalArrays.order(A.f.pool); ord[A.f[j]]`.

Reply:
Does that accomplish a faster lookup than findfirst(array,value)? I'm just
using the levels array to look up the index of that level.

Reply:
Indexing an array is (much) faster than a linear search, especially if levels are e.g. strings. Whether the difference matters or not depends on `n`.

Reply:
Okay I just didn't know that is what `CategoricalArrays.order` builds.

`n` here can be in the thousands, and this is called once for each row in the data. So, yes, worthwhile. Thanks!